lib/game.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Game do
+  @new_game %{
+    :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
+    :win? => false,
+    :player_1_moves => [],
+    :player_2_moves => [],
+    :player_1_mark => "X",
+    :player_2_mark => "O",
+    :current_player => 1,
+    :player_1_type => :human,
+    :player_2_type => :human,
+    :board => %{
+      1 => "1",
+      2 => "2",
+      3 => "3",
+      4 => "4",
+      5 => "5",
+      6 => "6",
+      7 => "7",
+      8 => "8",
+      9 => "9"
+    }
+  }
+
```

Comment:
I'm wondering if splitting the players into their own maps would be useful and clean things up...
Not requesting this as a change, just _spitballing_ but...

If you had:
```
@new_player %{
 :moves => [],
 :type => :unset,
 :mark => ""
}
```
and then something like
```
  @new_game %{
    :available_moves => [1, 2, 3, 4, 5, 6, 7, 8, 9],
    :win? => false,
    :players => [
      @new_player,
      @new_player
    ],
    :board => %{
      1 => "1",
      2 => "2",
      3 => "3",
      4 => "4",
      5 => "5",
      6 => "6",
      7 => "7",
      8 => "8",
      9 => "9"
    }
  }
```

I wonder if that might enable removing/simplifying some functions. Because it removes the need for stuff to know the current player. The current player could be determined by the first player in the `game.players`...
**examples below** (didn't test these so there might be some issues if you just copy and paste but you get the idea)
```
def list_moves(game), do: hd(game.players).moves

def switch_players(%{players: [current, next]} = game), do: %{game | players: [next, current]}

def assign_player_mark(marker, %{players: [player, other_player]} = game), do: %{game | players: [%{player | marker: marker}, other_player]}
```

Reply:
I like it! It offers another way to make an OOP style change to the program. Right now this would provide better organization, but I think it may take a few hours to make sure that all of the functions are rewritten to work correctly. If I was on a team, I would say we are messing around with the core logic and it would 100% have cascading impacts on anyone that is writing branches. I'd estimate that it would take 1/2 a day optimistically to get this working, with a full day being the worst case. 

Do you want this added as a tweak?

Reply:
No, was just spitballing & providing guidance for stuff to try.

Reply:
Good shout considering the downstream impacts of making the change 👍